RecoEgamma/EgammaTools/plugins/EGRegressionModifierV1.cc

```diff
@@ -0,0 +1,667 @@
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+
+#include "CondFormats/DataRecord/interface/GBRDWrapperRcd.h"
+#include "CondFormats/EgammaObjects/interface/GBRForestD.h"
+#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
+#include "CondFormats/EgammaObjects/interface/GBRForest.h"
+
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+
+#include "RecoEgamma/EgammaTools/interface/EcalClusterLocal.h"
+
+#include <vdt/vdtMath.h>
+
+namespace {
+  const edm::InputTag empty_tag;
+}
+
+#include <unordered_map>
+
+class EGRegressionModifierV1 : public ModifyObjectValueBase {
+public:
+  typedef edm::EDGetTokenT<edm::ValueMap<float> > ValMapFloatToken;
+  typedef edm::EDGetTokenT<edm::ValueMap<int> > ValMapIntToken;
+  typedef std::pair<edm::InputTag, ValMapFloatToken> ValMapFloatTagTokenPair;
+  typedef std::pair<edm::InputTag, ValMapIntToken> ValMapIntTagTokenPair;
+
+  struct electron_config {
+    edm::InputTag electron_src;
+    edm::EDGetTokenT<edm::View<pat::Electron> > tok_electron_src;
+    std::unordered_map<std::string, ValMapFloatTagTokenPair> tag_float_token_map;
+    std::unordered_map<std::string, ValMapIntTagTokenPair> tag_int_token_map;
+
+    std::vector<std::string> condnames_mean_50ns;
+    std::vector<std::string> condnames_sigma_50ns;
+    std::vector<std::string> condnames_mean_25ns;
+    std::vector<std::string> condnames_sigma_25ns;
+    std::string condnames_weight_50ns;
+    std::string condnames_weight_25ns;
+  };
+
+  struct photon_config {
+    edm::InputTag photon_src;
+    edm::EDGetTokenT<edm::View<pat::Photon> > tok_photon_src;
+    std::unordered_map<std::string, ValMapFloatTagTokenPair> tag_float_token_map;
+    std::unordered_map<std::string, ValMapIntTagTokenPair> tag_int_token_map;
+
+    std::vector<std::string> condnames_mean_50ns;
+    std::vector<std::string> condnames_sigma_50ns;
+    std::vector<std::string> condnames_mean_25ns;
+    std::vector<std::string> condnames_sigma_25ns;
+  };
+
+  EGRegressionModifierV1(const edm::ParameterSet& conf);
+  ~EGRegressionModifierV1() override {};
+    
+  void setEvent(const edm::Event&) final;
+  void setEventContent(const edm::EventSetup&) final;
+  void setConsumes(edm::ConsumesCollector&) final;
+  
+  void modifyObject(reco::GsfElectron&) const final;
+  void modifyObject(reco::Photon&) const final;
+  
+  // just calls reco versions
+  void modifyObject(pat::Electron&) const final; 
+  void modifyObject(pat::Photon&) const final;
+
+private:
+  electron_config e_conf;
+  photon_config   ph_conf;
+  std::unordered_map<unsigned,edm::Ptr<reco::GsfElectron> > eles_by_oop; // indexed by original object ptr
+  std::unordered_map<unsigned,edm::Handle<edm::ValueMap<float> > > ele_vmaps;
```

Comment:
please switch to the same style of local variables to use trailing underscore in the name